AxonDeepSeg/apply_model.py

```diff
@@ -1,73 +1,166 @@
-    if no_patch:
-        options["no_patch"] = no_patch
-        logger.warning("The 'no-patch' option was selected for segmentation. "\
-                       "Please note that it may not be suitable with large images depending on computer RAM capacity.")
-    else:
-        logger.warning("The 'no-patch' option was not selected for segmentation. "\
-                       "Please note that this option could potentially produce better results but may not be suitable "\
-                       "with large images depending on computer RAM capacity.")
-    if overlap_value:
-        # When both no_patch and overlap_value are used, the no_patch option supersedes the overlap_value
-        # and a warning will be issued by ivadomed while segmenting without patches.
-        options["overlap_2D"] = overlap_value
-    elif not no_patch:
-        # Default overlap is used only without the no_patch option.
-        options["overlap_2D"] = [default_overlap, default_overlap]
-
-    # IVADOMED automated segmentation
-    nii_lst, _ = imed_inference.segment_volume(str(path_model), input_filenames, gpu_id=gpu_id, options=options)
-    
-    target_lst = [str(axon_suffix), str(myelin_suffix)]
+    Errors
+    ------
+    ValueError
+        If the class value is not found in the raw prediction.
+    ValueError
+        If the raw nnunet prediction file does not end with '_seg-nnunet.png'.
 
-    imed_inference.pred_to_png(nii_lst, target_lst, str(Path(input_filenames[0]).parent / Path(input_filenames[0]).stem))
-    if verbosity_level >= 1:
-        print(Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axonmyelin_suffix)))
+    Returns
+    -------
+    new_fname : str
+        Path to the extracted class mask saved.
+    '''
+
+    pred_path = ads_utils.convert_path(pred_path)
+    if not np.any(pred == class_value):
+        raise ValueError(f'Class value {class_value} not found in the raw prediction.')
+    elif not pred_path.name.endswith(str(nnunet_suffix)):
+        raise ValueError(f'Raw nnunet pred file does not end with "{nnunet_suffix}".')
     
-    merge_masks(
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axon_suffix)),
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(myelin_suffix)), 
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axonmyelin_suffix))
-        )
+    extraction = np.zeros_like(pred)
+    extraction[pred == class_value] = intensity['binary']
+    new_fname = str(pred_path).replace(str(nnunet_suffix), f'_seg-{class_name}.png')
+    ads_utils.imwrite(new_fname, extraction)
+
+    return new_fname
+
+def axon_segmentation(
+                    path_inputs: List[Path],
+                    path_model: Path,
+                    model_type: Literal['light', 'ensemble']='light',
+                    gpu_id: int=-1,
+                    verbosity_level: int=0,
+                    ) -> NoReturn:
```

Comment:
Note that this function creates 3 JSON files everytime it is called: _dataset.json_, _plans.json_ and _predict_from_raw_data_args.json_. Maybe we should delete these?